tasks/rubies.yml

```diff
@@ -62,16 +62,24 @@
   when: not '--user-install' in rvm1_install_flags and rvm1_bundler_install
   with_items: '{{ rvm1_symlink_bundler_binaries }}'
 
-- name: List installed versions of Ruby (for deletion)
+- name: List installed ruby versions (for deletion)
   command: '{{ rvm1_rvm }} list strings'
   changed_when: False
   register: rvm1_list_installed_rubies
-  when: rvm1_delete_ruby is defined and rvm1_delete_ruby != ''
+  when: rvm1_delete_unmanaged_rubies|bool or (rvm1_delete_ruby is defined and rvm1_delete_ruby != '')
 
-- name: Delete ruby if relevant
+- name: Delete unmanaged ruby versions
+  command: '{{ rvm1_rvm }} remove {{ item }}'
+  register: rvm_delete_rubies_command_result
+  changed_when: "'#removing' in rvm_delete_rubies_command_result.stdout"
+  when: rvm1_delete_unmanaged_rubies|bool
+  with_items: '{{ rvm1_list_installed_rubies.stdout.splitlines() | difference(rvm1_rubies|default([])) }}'
```

Comment:
I think `loop` is now preferred over `with_items`.
https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html

And maybe use `rvm1_list_installed_rubies.stdout_lines` instead of `rvm1_list_installed_rubies.stdout.splitlines()`.

Reply:
I copy-pasted `rvm1_list_installed_rubies.stdout.splitlines()` from the existing tasks, but thank you for giving this hint. I'll change it in both tasks.

About `loop` versus `with_items`: Since all the "looping" tasks in this role are still using the original `with_items` style, I didn't want to mix both syntaxes. I won't change it for the moment, and would propose to convert all the relevant tasks via a single ad-hoc commit.

Reply:
See fada459a67fc67e0d5a11dee507d2ab76ecc7a3a